### ISSUE TITLE
Update FSDPPrecision to support `16-true` precision setting

### ISSUE THREAD

**Issue by awaelchli** [Lightning-AI/lightning#17657]
### Description & Motivation

We recently added `true` precision support in Fabric:
#17287 
#17576 

The FSDPPrecision plugin needs an update to support the strings "16-true" and "bf16-true". 

### Pitch

Update the plugin's code by following the same pattern as in #17576.

cc @borda @justusschock @awaelchli @carmocca

**Comment by carmocca** [comment]
I think https://github.com/Lightning-AI/lightning/pull/17670 fixes this issue too

**Comment by speediedan** [comment]
You guys may have observed this internally already but if not I thought I'd share that though #17670 partially addresses this issue, patching of at least the following two sections was required to enable FSDP `16-true` support for the [downstream package](https://github.com/speediedan/finetuning-scheduler) I maintain:  https://github.com/Lightning-AI/lightning/blob/0cc458e237c19a725d57ed33b3abe90dd7a0d3a4/src/lightning/pytorch/plugins/precision/fsdp.py#L36-L43 and https://github.com/Lightning-AI/lightning/blob/0cc458e237c19a725d57ed33b3abe90dd7a0d3a4/src/lightning/pytorch/trainer/connectors/accelerator_connector.py#L542-L544

Thanks for all your continued work making Lightning better every day!
@awaelchli @carmocca

**Comment by awaelchli** [comment]
@speediedan https://github.com/Lightning-AI/lightning/pull/17670 was a different issue. We definitely want to bring support for 16-true to the Lightning Trainer (#17609) and with it also in the special FSDP precision plugin. What you point out is the change required to make the plugin accept the strings, and that's basically what I opened this ticket for, just for the Fabric side of things. But yes, it applies to both Fabric and PyTorch :)

**Comment by speediedan** [comment]
Awesome, figured it was all planned but thanks for thorough (as always!) clarification @awaelchli! Reminds me I'm looking forward to contributing/collaborating with you guys more in the future!
